### PR TITLE
query request pool

### PR DESCRIPTION
**Pull request by songwongtp**
Fixed: #XXXX

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: proto/oracle/v1/query.proto [hunk 1]
```diff
@@ -70,6 +71,13 @@ service Query {
       returns (QueryRequestPriceResponse) {
     option (google.api.http).post = "/oracle/request_prices";
   }
+
+  // RequestPool queries the request pool information corresponding to the given
+  // port, channel, and request key.
+  rpc RequestPool(QueryRequestPoolRequest) returns (QueryRequestPoolResponse) {
+    option (google.api.http).get =
+        "/oracle/request_pool/{request_key}/{port_id}/{channel_id}";
```
Comment (Benzbeeb) [inline proto/oracle/v1/query.proto:79]: "/oracle/v1/request_pool/{request_key}/{port_id}/{channel_id}";

**Review by Benzbeeb — CHANGES_REQUESTED** [review]

File: proto/oracle/v1/query.proto (hunk 1 above)
Comment (songwongtp) [reply proto/oracle/v1/query.proto]: d4c2959

File: proto/oracle/v1/query.proto [hunk 2]
```diff
@@ -226,3 +234,24 @@ message QueryRequestPriceResponse {
...
+// QueryRequestPoolResponse is response type for the Query/RequestPool RPC
+// method.
+message QueryRequestPoolResponse {
+  // RequestPoolAddress is an address of a request pool
+  string request_pool_address = 1;
+  repeated cosmos.base.v1beta1.Coin balance = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+}
```
Comment (ntchjb) [inline proto/oracle/v1/query.proto:257]: New line EOF

File: proto/oracle/v1/query.proto [hunk 3]
```diff
@@ -70,6 +71,13 @@ service Query {
       returns (QueryRequestPriceResponse) {
     option (google.api.http).post = "/oracle/request_prices";
   }
+
+  // RequestPool queries the request pool information corresponding to the given
+  // port, channel, and request key.
+  rpc RequestPool(QueryRequestPoolRequest) returns (QueryRequestPoolResponse) {
+    option (google.api.http).get =
+        "/oracle/v1/request_pool/{request_key}/{port_id}/{channel_id}";
```
Comment (ntchjb) [inline proto/oracle/v1/query.proto:79]: `request_key` may be considered a secret, so I'm not sure whether it is appropriate to put `request_key` in HTTP URL.

Is it possible to not include the `request_key` in the URL but still include in `QueryRequestPoolRequest`?

File: go.mod [hunk 4]
```diff
@@ -24,6 +24,7 @@ require (
 	github.com/poy/onpar v1.1.2 // indirect
 	github.com/prometheus/client_golang v1.8.0
 	github.com/rakyll/statik v0.1.7
+	github.com/regen-network/cosmos-proto v0.3.1 // indirect
```
Comment (ntchjb) [inline go.mod:27]: Run `go mod tidy` may remove this dependency.

**Review by ntchjb — CHANGES_REQUESTED** [review]

File: go.mod (hunk 4 above)
Comment (songwongtp) [reply go.mod]: 6f1537f

File: proto/oracle/v1/query.proto (hunk 2 above)
Comment (songwongtp) [reply proto/oracle/v1/query.proto]: 6f1537f

File: proto/oracle/v1/query.proto (hunk 3 above)
Comment (songwongtp) [reply proto/oracle/v1/query.proto:79]: What about using the sender address on the counterparty chain as the request key?

**Review by taobun — APPROVED** [review]

**Review by ntchjb — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]

**Pull request merged**